haystack/components/classifiers/zero_shot_document_classifier.py

```
@@ -0,0 +1,225 @@
+        """
+        Deserializes the component from a dictionary.
+
+        :param data:
+            Dictionary to deserialize from.
+        :returns:
+            Deserialized component.
+        """
+        deserialize_secrets_inplace(data["init_parameters"], keys=["token"])
+        if data["init_parameters"].get("huggingface_pipeline_kwargs") is not None:
+            deserialize_hf_model_kwargs(data["init_parameters"]["huggingface_pipeline_kwargs"])
+        return default_from_dict(cls, data)
+
+    @component.output_types(documents=Dict[str, List[Document]])
+    def run(self, documents: List[Document], batch_size: int = 1):
+        """
+        This method classifies the documents based on the provided labels and adds it to their metadata.
+
+        Documents are updated in place. The classification results are stored in the `classification` dict within
+        each document's metadata. If `multi_label` is set to `True`, the scores for each label are available under
+        the `details` key within the `classification` dictionary.
+
+        :param documents:
+            Documents to process.
+        :param batch_size:
+            Batch size used for processing the content in each document.
+        """
+
+        if self.pipeline is None:
+            raise RuntimeError(
+                "The component TransformerZeroShotDocumentClassifier wasn't warmed up. "
+                "Run 'warm_up()' before calling 'run()'."
+            )
+
+        if not isinstance(documents, list) or documents and not isinstance(documents[0], Document):
+            raise TypeError(
+                "DocumentLanguageClassifier expects a list of Document as input. "
+                "In case you want to classify a text, please use the TextLanguageClassifier."
+            )
+
+        texts = [
+            doc.content if self.classification_field is None else doc.meta[self.classification_field]
```

The given metadata field could be missing, so let's wrap this in exception handling and raise an error if the field is missing.